src/pip/_internal/models/index.py

```diff
@@ -2,14 +2,19 @@
 
 
 class Index(object):
-    def __init__(self, url):
+    def __init__(self, url, link_source=''):
         self.url = url
         self.netloc = urllib_parse.urlsplit(url).netloc
         self.simple_url = self.url_to_path('simple')
         self.pypi_url = self.url_to_path('pypi')
 
+        # This part of a temporary hack used to block installs of PyPI packages
+        # which depend on external urls only necessary until PyPI can block
+        # such packages themselves
+        self.link_source = link_source
```

Comment:
Call this something else? `link_source` seems very ambiguous and is completely out of context in this file.

Possibly `file_storage_domain`?